File: Java/src/main/csci/ooad/grad/Entities/Cat.java
package csci.ooad.grad.Entities;

public class Cat extends Entity implements CanMeow {

    public Cat(String name) {
        super(name);

        setNoise("meow");
    }

    public String meow(Entity annoyedAt) {
        return getNoise() + " to " + annoyedAt.getName();
    }

}

File: TypeScript/src/Entities/Cat.ts
import { Entity } from "./Entity";
import { CanMeow } from "./CanMeow";

export class Cat extends Entity implements CanMeow {
    constructor(name: string) {
        super(name);
        this.noise = "meow";
    }

    public meow(annoyedAt: Entity): string {
        return `${this.getNoise()} to ${annoyedAt.getName()}`;
    }
}
